diff --git a/boost/optional/optional.hpp b/boost/optional/optional.hpp
index 07c4bd3..c021319 100644
--- a/boost/optional/optional.hpp
+++ b/boost/optional/optional.hpp
@@ -1227,9 +1227,9 @@ get_pointer ( optional<T>& opt )
 }
 
 // Forward declaration to prevent operator safe-bool from being used.
-template<class CharType, class CharTrait, class T>
-std::basic_ostream<CharType, CharTrait>&
-operator<<(std::basic_ostream<CharType, CharTrait>& out, optional<T> const& v);
+// template<class CharType, class CharTrait, class T>
+// std::basic_ostream<CharType, CharTrait>&
+// operator<<(std::basic_ostream<CharType, CharTrait>& out, optional<T> const& v);
 
 // optional's relational operators ( ==, !=, <, >, <=, >= ) have deep-semantics (compare values).
 // WARNING: This is UNLIKE pointers. Use equal_pointees()/less_pointess() in generic code instead.
 