diff --git a/libs/context/src/asm/jump_arm_aapcs_macho_gas.S b/libs/context/src/asm/jump_arm_aapcs_macho_gas.S
index 6184875..e651ff8 100644
--- a/libs/context/src/asm/jump_arm_aapcs_macho_gas.S
+++ b/libs/context/src/asm/jump_arm_aapcs_macho_gas.S
@@ -91,13 +91,13 @@ _jump_fcontext:
     add  sp, sp, #64
 
     @ restore SjLj handler
-    pop  v1
+    pop  {v1}
     @ store SjLj handler in TLS
     str  v1, [v2, #72]
 
     @ use third arg as return value after jump
     @ and as first arg in context function
     mov  a1, a3
-    
+
     @ restore v1-V8,LR,PC
     pop {v1-v8,lr,pc}
