--- a/src/tool_operate.c
+++ b/src/tool_operate.c
@@ -1795,7 +1795,11 @@ CURLcode operate(struct GlobalConfig *config, int argc, argv_item_t argv[])
         tool_help();
       /* Check if we were asked for the manual */
       else if(res == PARAM_MANUAL_REQUESTED)
+#ifdef USE_MANUAL
         hugehelp();
+#else
+        tool_help();
+#endif
       /* Check if we were asked for the version information */
       else if(res == PARAM_VERSION_INFO_REQUESTED)
         tool_version_info();
